src/DeploymentManager/DeploymentManager/Commands/RemoveServiceUnit.cs

```diff
@@ -0,0 +1,261 @@
+            Mandatory = true, 
+            ParameterSetName = RemoveServiceUnit.ByTopologyObjectAndServiceNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The name of the service unit.")]
+        [Parameter(
+            Position = 3,
+            Mandatory = true, 
+            ParameterSetName = RemoveServiceUnit.ByTopologyResourceIdAndServiceNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The name of the service unit.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true, 
+            ParameterSetName = DeploymentManagerBaseCmdlet.ResourceIdParamSetName,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource identifier.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true, 
+            ParameterSetName = DeploymentManagerBaseCmdlet.InputObjectParamSetName,
+            ValueFromPipeline = true,
+            HelpMessage = "Service unit resource object.")]
+        [ValidateNotNullOrEmpty]
+        public PSServiceUnitResource ServiceUnit { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true, 
+            ParameterSetName = RemoveServiceUnit.ByTopologyObjectAndServiceNameParameterSet,
+            HelpMessage = "The service topology object in which the service unit should be created.")]
+        [ValidateNotNullOrEmpty]
+        public PSServiceTopologyResource ServiceTopology { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true, 
+            ParameterSetName = RemoveServiceUnit.ByTopologyResourceIdAndServiceNameParameterSet,
+            HelpMessage = "The service topology resource identifier in which the service unit should be created.")]
+        [ValidateNotNullOrEmpty]
+        public string ServiceTopologyResourceId { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true, 
+            ParameterSetName = RemoveServiceUnit.ByServiceObjectParameterSet,
+            HelpMessage = "The service object in which the service unit should be created.")]
+        [ValidateNotNullOrEmpty]
+        public PSServiceResource Service { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true, 
+            ParameterSetName = RemoveServiceUnit.ByServiceResourceIdParamSet,
+            HelpMessage = "The service resource identifier in which the service unit should be created.")]
+        [ValidateNotNullOrEmpty]
+        public string ServiceResourceId { get; set; }
+
+
+        [Parameter(
+            Mandatory = false, 
+            HelpMessage = "Do not ask for confirmation.")]
+        public SwitchParameter Force { get; set; }
```

Comment:
@deveshguha same comment about verifying whether or not this is needed

Reply:
Was made to add these in the earlier version as these are destructive operations.

Reply:
@deveshguha same comment as above

Reply:
Removed